x-pack/plugins/search_inference_endpoints/public/components/inference_flyout_wrapper_component.tsx

```diff
@@ -88,7 +94,7 @@ export const InferenceFlyoutWrapperComponent: React.FC<InferenceFlyoutWrapperCom
     async (inferenceId: string, taskType: InferenceTaskType, modelConfig: ModelConfig) => {
       setIsCreateInferenceApiLoading(true);
       try {
-        await createInferenceEndpointMutation.mutateAsync({ inferenceId, taskType, modelConfig });
+        createInferenceEndpointMutation.mutateAsync({ inferenceId, taskType, modelConfig });
```

Comment:
If you remove the `await` here the `catch` now does nothing. You will get an unhandled promise rejection instead.

If you want to remove the `await` then you should remove the `try/catch` as well and replace it with explicitly handling the Promise.

```suggestion
        createInferenceEndpointMutation.mutateAsync({ inferenceId, taskType, modelConfig })
          .catch((err) => {
            const errorObj = extractErrorProperties(error);
            setInferenceAddError(errorObj.message);
          })
          .finally(() => {
            setIsCreateInferenceApiLoading(false);
          });
```

I'm not saying you _should_ do that, just explainingg the result of removing the `await` here.

Reply:
Updated